src/Facets/WhitelistManagerFacet.sol

```diff
@@ -0,0 +1,232 @@
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+        _addToWhitelist(_contractAddress);
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function batchAddToWhitelist(address[] calldata _addresses) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+        uint256 length = _addresses.length;
+
+        for (uint256 i = 0; i < length; ++i) {
+            _addToWhitelist(_addresses[i]);
+        }
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function removeFromWhitelist(address _address) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+        _removeFromWhitelist(_address);
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function batchRemoveFromWhitelist(address[] calldata _addresses) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+        uint256 length = _addresses.length;
+        for (uint256 i = 0; i < length; ++i) {
+            _removeFromWhitelist(_addresses[i]);
+        }
+    }
+
+    /// @notice Adds or removes a specific function selector to/from the allowlist.
+    /// @param _selector The function selector to allow or disallow.
+    /// @param _approval Whether the function selector should be allowed.
+    function setFunctionApprovalBySelector(
+        bytes4 _selector,
+        bool _approval
+    ) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+        _setFunctionApproval(_selector, _approval);
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function batchSetFunctionApprovalBySelector(
+        bytes4[] calldata _selectors,
+        bool _approval
+    ) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+        uint256 length = _selectors.length;
+        for (uint256 i = 0; i < length; ++i) {
+            _setFunctionApproval(_selectors[i], _approval);
+        }
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function isFunctionApproved(
+        bytes4 _selector
+    ) external view returns (bool approved) {
+        return LibAllowList.selectorIsAllowed(_selector);
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function getWhitelistedAddresses()
+        external
+        view
+        returns (address[] memory addresses)
+    {
+        return LibAllowList.getAllowedContracts();
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function isAddressWhitelisted(
+        address _address
+    ) external view returns (bool approved) {
+        return LibAllowList.contractIsAllowed(_address);
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function getApprovedFunctionSelectors()
```

Comment:
nitpick: 
would probably be more coherent to name this function "getWhitelistedFunctionSelectors", or short "getWhitelistedFunctionSelectors"

but I am also fine to leave it as-is